### PR TITLE
DROOLS-2584: [DMN Designer] Type Ref select box options are not sorted

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2584

### CODE REVIEW

**Comment by jomarko** [comment]
jenkins please retest this

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/TypePickerWidget.java [hunk 1]
```diff
@@ -52,6 +55,10 @@
 public class TypePickerWidget extends Composite implements HasValue<String>,
                                                            HasEnabled {
 
+    static final Comparator<BuiltInType> BUILT_IN_TYPE_COMPARATOR = (o1, o2) -> o1.getName().compareTo(o2.getName());
+
+    static final Comparator<ItemDefinition> ITEM_DEFINITION_COMPARATOR = (o1, o2) -> o1.getName().getValue().compareTo(o2.getName().getValue());
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/TypePickerWidget.java]: Thanks @jomarko your PR has been merged and incorporated into this PR!

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**